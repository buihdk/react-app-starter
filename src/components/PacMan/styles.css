@keyframes dot-1 {
  0% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(-45deg);
  }
  100% {
    transform: rotate(0deg);
  }
}
@keyframes dot-2 {
  0% {
    transform: rotate(180deg);
  }
  50% {
    transform: rotate(225deg);
  }
  100% {
    transform: rotate(180deg);
  }
}
.dot > div:nth-child(2) {
  transform: translate(-15px, 0);
}
.dot > div:nth-child(2) div {
  position: absolute;
  top: 40px;
  left: 40px;
  width: 120px;
  height: 60px;
  border-radius: 120px 120px 0 0;
  background: #f8b26a;
  animation: dot-1 1s linear infinite;
  transform-origin: 60px 60px;
}
.dot > div:nth-child(2) div:nth-child(2) {
  animation: dot-2 1s linear infinite;
}
.dot > div:nth-child(2) div:nth-child(3) {
  transform: rotate(-90deg);
  animation: none;
}
@keyframes dot-3 {
  0% {
    transform: translate(190px, 0);
    opacity: 0;
  }
  20% {
    opacity: 1;
  }
  100% {
    transform: translate(70px, 0);
    opacity: 1;
  }
}
.dot > div:nth-child(1) {
  display: block;
}
.dot > div:nth-child(1) div {
  position: absolute;
  top: 92px;
  left: -8px;
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background: #e15b64;
  animation: dot-3 1s linear infinite;
}
.dot > div:nth-child(1) div:nth-child(1) {
  animation-delay: -0.67s;
}
.dot > div:nth-child(1) div:nth-child(2) {
  animation-delay: -0.33s;
}
.dot > div:nth-child(1) div:nth-child(3) {
  animation-delay: 0s;
}
.spinner-pac-man {
  width: 200px;
  height: 200px;
  display: inline-block;
  overflow: hidden;
  background: #ffffff;
}
.dot {
  width: 100%;
  height: 100%;
  position: relative;
  transform: translateZ(0) scale(1);
  backface-visibility: hidden;
  transform-origin: 0 0;
}
.dot div {
  box-sizing: content-box;
}
