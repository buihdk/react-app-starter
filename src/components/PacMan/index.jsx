import React from 'react';

import './styles.css';

const PacMan = () => (
  <div className="spinner-pac-man">
    <div className="dot">
      <div>
        <div />
        <div />
        <div />
      </div>
      <div>
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
);

export default PacMan;
