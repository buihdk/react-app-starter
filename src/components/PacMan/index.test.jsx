import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import PacMan from '.';

describe('PacMan', () => {
  const wrapper = shallow(<PacMan />);

  test('renders without crashing', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
