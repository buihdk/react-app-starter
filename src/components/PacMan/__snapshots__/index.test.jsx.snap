// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PacMan renders without crashing 1`] = `
<div
  className="spinner-pac-man"
>
  <div
    className="dot"
  >
    <div>
      <div />
      <div />
      <div />
    </div>
    <div>
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
